compact/tracking/mpgd_barrel.xml

```diff
@@ -21,70 +21,91 @@
-        <module_component name="CuStrips" thickness="MMCuStrip_thickness" material="Copper" width="InnerMPGDBarrelStave_width" length="InnerMPGDBarrel_length"/>
-        <module_component name="PCB" thickness="MMPCB_thickness" material="Fr4" width="InnerMPGDBarrelStave_width" length="InnerMPGDBarrel_length"/>
-        <module_component name="CuGround" thickness="MMCuGround_thickness" material="Copper" width="InnerMPGDBarrelStave_width" length="InnerMPGDBarrel_length"/>
-        <module_component name="KaptonOverlay" thickness="MMKaptonOverlay_thickness" material="Kapton" vis="TrackerSupportVis" width="InnerMPGDBarrelStave_width" length="InnerMPGDBarrel_length"/>
+        <module_component name="DriftCuGround" thickness="MMDriftCuGround_thickness" material="Copper" vis="TrackerModuleVis"/>
+        <module_component name="DriftKapton" thickness="MMDriftKapton_thickness" material="Kapton" vis="TrackerVis" />
+        <module_component name="DriftCuElectrode" thickness="MMDriftCuElectrode_thickness" material="Copper" vis="TrackerVis"/>
+        <module_component name="GasGap" thickness="MMGasGap_thickness" material="Ar90IsoButane" sensitive="True" vis="TrackerMPGDGasVis" />
+        <module_component name="Mesh" thickness="MMMesh_thickness" material="MMGAS_InoxForMesh" vis="TrackerVis"/>
+        <module_component name="Fudge" thickness="MMFudgeInnerMPGDBarrel_thickness" material="Kapton" vis="TrackerVis"/>
+        <module_component name="Gas" thickness="MMGas_thickness" material="Ar90IsoButane" vis="TrackerMPGDGasVis"/>
+        <module_component name="ResistiveStrips" thickness="MMResistiveStrip_thickness" material="MMGAS_ResistivePaste" vis="TrackerVis"/>
+        <module_component name="KaptonStrips" thickness="MMKaptonStrip_thickness" material="Kapton" vis="TrackerVis" />
+        <module_component name="CuStrips" thickness="MMCuStrip_thickness" material="Copper" vis="TrackerVis" />
+        <module_component name="PCB" thickness="MMPCB_thickness" material="Fr4" vis="TrackerVis"/>
+        <module_component name="CuGround" thickness="MMCuGround_thickness" material="Copper" vis="TrackerVis"/>
+        <module_component name="KaptonOverlay" thickness="MMKaptonOverlay_thickness" material="Kapton" vis="TrackerMPGDVis"/> <!-- Instead of "TrackerSupportVis" in order to highligh frames -->
       </module>
-      <layer module="InnerMPGDBarrel_Mod1" id="1" vis="TrackerMMGASLayerVis">
+
+      <layer module="InnerMPGDBarrel_Mod1" id="1" vis="InvisibleWithDaughters">
         <barrel_envelope
           inner_r="InnerMPGDBarrelLayer_rmin"
           outer_r="InnerMPGDBarrelLayer_rmax"
           z_length="InnerMPGDBarrelLayer_length"
           z0="InnerMPGDBarrel_zoffset"/>
         <layer_material surface="inner" binning="binPhi,binZ" bins0="MPGDBarrelStave_count" bins1="100" />
-        <layer_material surface="outer" binning="binPhi,binZ" bins0="MPGDBarrelStave_count" bins1="100" />
```

Comment:
@ShujieL may confirm, but we have both inner and outer surfaces to ensure that material maps can allow for both approaches.

Reply:
@wdconinc I think that is correct. I remember when going over material mapping with @ShujieL she mentioned she wanted to add two surfaces for the material to obtain both approaches

Reply:
I now understand that there is something wrong here, concerning "<layer_material/>", which I have overlooked. "MPGDBarrelStave_count" is set =8 having in mind that whatever do-loop including it must have it as its stopping  limit. => Therefore, something like bins0="0" bins1="MPGDBarrelStave_count - 1" />. To makes things clearer: CyMBaL is subdivided into 8 tiles of cylindrical shape along phi, whereas previously the barrel shape along phi was a polygon w/ 100 sides. But then I have to admit that I do not what this <layer_material/> line of instruction is supposed to do: I do not know myself how to modify it. And I would be willing to learn, btw. Where in the software is this piece of info used, e.g.?

Reply:
I think adding the deleted line back in should be ok. These are related to ACTS and tell it how to map the material onto the inner and outer layer surface. "binning" give the bin types, e.g. phi,Z. bins0 is number of phi bins and bins1 is for bins in z that the material is mapped to. You can see some explanation of the "inner" and "outer" surfaces here:https://acts.readthedocs.io/en/v7.0.0/core/material.html 

and some of parameter meanings here: https://acts.readthedocs.io/en/latest/plugins/dd4hep.html#possible-parameters-acts-will-interpret

Reply:
I implemented the duplication (inner += outer). See commit #7403586 ("https://github.com/eic/epic/commit/7403586").
But I'm not sure  that the settings of "bins(0|1)" are correct. What binning is here at stake?: I haven't understood. And the setting of "bins0" was originally set =128 (via MPGDBarrelStave_count=128). Does it make sense that it now goes from 128 to only 8? Anyway that does not seem to have any impact on the distributions I am monitoring (Sim and Rec hits eicrecon log).

Reply:
My understanding of this is that "inner_r" defined in the <barrel_envelope> tag creates an ACTS surface cylinder surface with radius of inner_r and length "z_length", which ACTS projects near by materials onto. The material on this cylindrical surface is binned in phi and z, with Bins0 = number of phi bins, and Bins1 = the number of Z bins used. There is a second surface, "outer_r", that does a similar thing to create an outer material surface. In the previous barrel, a cylinder was approximated using 128 staves, and the phi binning was defined based on the number of staves. For your updated cylinder, I think 8 phi bins would work, e.g. 1 phi bin per tile. A finer binning could be used and would result in a finer material mapping to these surfaces.

Reply:
If you use 8 phi bins for a detector with 8-fold symmetry, then your bins are going to be filled with the same value and you might as well use 1 bin... What you are trying to account for here is the variation in material within these 2pi/8 bins: if a track is closer to the edge of a tile, the projection of material on a cylinder will be different by 1/cos(2pi/16) (if I did this correctly without a napkin), i.e. an 8% effect.

In some sense, you may want to think about the size of features between tiles, e.g. areas of overlap or gaps. The angular dimension of those features are what should define the bin size of the material map approximation.

Reply:
I increased the binning so as to be sensitive to ~2 cm, i.e. the cumulated width of two neighboring detector frames.